Can't install using yay

**Describe the bug**
I have followed the guide of installing in Arch using yay (also tried with paru) but it doesn't work. It always fails when importing some libraries

**To Reproduce**
Simply the command: `yay -S ags-hyprpanel-git`. Also tried compiling from the repo.

**Screenshots**
I don't have a screenshots but text
```
✘ [ERROR] Could not resolve "src/lib/constants/services"

    ../src/components/menus/network/wifi/WirelessAPs/helpers.ts:5:31:
      5 │ import { networkService } from 'src/lib/constants/services';
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "src/lib/constants/services" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "src/lib/utils"

    ../src/components/menus/network/wifi/WirelessAPs/helpers.ts:6:39:
      6 │ import { isPrimaryClick, Notify } from 'src/lib/utils';
        ╵                                        ~~~~~~~~~~~~~~~

  You can mark the path "src/lib/utils" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "src/lib/constants/services"

    ../src/components/menus/network/wifi/WirelessAPs/AccessPoint.tsx:6:31:
      6 │ import { networkService } from 'src/lib/constants/services';
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "src/lib/constants/services" as external to exclude it from the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "src/components/shared/Spinner"

    ../src/components/menus/network/wifi/WirelessAPs/AccessPoint.tsx:7:20:
      7 │ import Spinner from 'src/components/shared/Spinner';
```

**Desktop (please complete the following information):**
 - Distribution: Arch Linux installed yesterday
 - Window Manager/Desktop Environment: Hyprland

--- a/env.d.ts
+++ b/env.d.ts
@@ -1,4 +1,5 @@
 declare const SRC: string;
+declare const DATADIR: string;
 
 declare module 'inline:*' {
     const content: string;

--- a/meson.build
+++ b/meson.build
@@ -1,4 +1,9 @@
-project('hyprpanel')
+project(
+  'hyprpanel',
+  default_options: [
+    'prefix=/usr',
+  ],
+)
 
 prefix = get_option('prefix')
 bindir = prefix / get_option('bindir')
@@ -7,47 +12,21 @@ datadir = prefix / get_option('datadir') / meson.project_name()
 ags = find_program('ags', required: true)
 find_program('gjs', required: true)
 
-src_file_list_process = run_command(
-  'find',
-  'src',
-  '-type', 'f',
-  '(',
-  '-name', '*.ts',
-  '-o',
-  '-name', '*.tsx',
-  '-o',
-  '-name', '*.scss',
-  ')',
-)
-
-if src_file_list_process.returncode() != 0
-  error('Failed to find source files.')
-endif
-
-src_file_list = src_file_list_process.stdout().split('\n')
-
-all_sources = []
-
-foreach file : src_file_list
-  file_stripped = file.strip()
-  if file_stripped != ''
-    all_sources += meson.project_source_root() / file_stripped
-  endif
-endforeach
-
 custom_target(
   'hyprpanel_bundle',
-  input: all_sources,
+  input: files('app.ts'),
   command: [
     ags,
     'bundle',
     '--define', 'DATADIR="' + datadir + '"',
+    '--root', meson.project_source_root(),
     meson.project_source_root() / 'app.ts',
-    meson.project_name(),
+    '@OUTPUT@',
   ],
   output: meson.project_name() + '.js',
   install: true,
   install_dir: datadir,
+  build_always_stale: true,
 )
 
 configure_file(

--- a/meson.build
+++ b/meson.build
@@ -52,7 +52,7 @@ custom_target(
 
 configure_file(
   input: 'scripts/hyprpanel_launcher.sh.in',
-  output: 'hyprpanel',
+  output: meson.project_name(),
   configuration: {'DATADIR': datadir},
   install: true,
   install_dir: bindir,

--- a/meson.build
+++ b/meson.build
@@ -1,7 +1,8 @@
 project('hyprpanel')
 
-bindir = get_option('prefix') / get_option('bindir')
-datadir = get_option('prefix') / get_option('datadir') / 'hyprpanel'
+prefix = get_option('prefix')
+bindir = prefix / get_option('bindir')
+datadir = prefix / get_option('datadir') / meson.project_name()
 
 ags = find_program('ags', required: true)
 find_program('gjs', required: true)
@@ -40,11 +41,11 @@ custom_target(
   command: [
     ags,
     'bundle',
+    '--define', 'DATADIR="' + datadir + '"',
     meson.project_source_root() / 'app.ts',
-    '@OUTPUT@',
-    '--src', meson.project_source_root(),
+    meson.project_name(),
   ],
-  output: 'hyprpanel.js',
+  output: meson.project_name() + '.js',
   install: true,
   install_dir: datadir,
 )

--- a/package.json
+++ b/package.json
@@ -2,10 +2,7 @@
     "name": "hyprpanel",
     "version": "1.0.0",
     "description": "A customizable panel built for Hyprland.",
-    "main": "config.js",
-    "directories": {
-        "lib": "lib"
-    },
+    "main": "app.ts",
     "scripts": {
         "lint": "eslint --config .eslintrc.js .",
         "lint:fix": "eslint --config .eslintrc.js . --fix",
@@ -14,6 +11,9 @@
     "keywords": [],
     "author": "",
     "license": "MIT",
+    "dependencies": {
+        "astal": "/usr/share/astal/gjs"
+    },
     "devDependencies": {
         "@types/node": "^22.5.4",
         "@typescript-eslint/eslint-plugin": "^8.5.0",

--- a/scripts/hyprpanel_launcher.sh.in
+++ b/scripts/hyprpanel_launcher.sh.in
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-export HYPRPANEL_DATADIR="@DATADIR@"
+# export HYPRPANEL_DATADIR="@DATADIR@"
 
 if [ "$#" -eq 0 ]; then
     exec gjs -m "@DATADIR@/hyprpanel.js"

--- a/scripts/hyprpanel_launcher.sh.in
+++ b/scripts/hyprpanel_launcher.sh.in
@@ -1,7 +1,5 @@
 #!/bin/sh
 
-# export HYPRPANEL_DATADIR="@DATADIR@"
-
 if [ "$#" -eq 0 ]; then
     exec gjs -m "@DATADIR@/hyprpanel.js"
 else

--- a/src/lib/session.ts
+++ b/src/lib/session.ts
@@ -28,11 +28,13 @@ export function ensureFile(path: string): void {
     }
 }
 
+const dataDir = typeof DATADIR !== 'undefined' ? DATADIR : SRC;
+
 Object.assign(globalThis, {
     CONFIG: `${GLib.get_user_config_dir()}/hyprpanel/config.json`,
     TMP: `${GLib.get_tmp_dir()}/hyprpanel`,
     USER: GLib.get_user_name(),
-    SRC_DIR: DATADIR,
+    SRC_DIR: dataDir,
 });
 
 ensureDirectory(TMP);

--- a/src/lib/session.ts
+++ b/src/lib/session.ts
@@ -32,7 +32,7 @@ Object.assign(globalThis, {
     CONFIG: `${GLib.get_user_config_dir()}/hyprpanel/config.json`,
     TMP: `${GLib.get_tmp_dir()}/hyprpanel`,
     USER: GLib.get_user_name(),
-    SRC_DIR: GLib.getenv('HYPRPANEL_DATADIR') ?? SRC,
+    SRC_DIR: DATADIR,
 });
 
 ensureDirectory(TMP);

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -19,8 +19,6 @@
         "types": [],
         "experimentalDecorators": true,
         "jsx": "react-jsx",
-        "jsxImportSource": "astal/gtk3",
-        "paths": {
-        }
+        "jsxImportSource": "astal/gtk3"
     }
 }

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -19,11 +19,8 @@
         "types": [],
         "experimentalDecorators": true,
         "jsx": "react-jsx",
-        "jsxImportSource": "/usr/share/astal/gjs/gtk3",
+        "jsxImportSource": "astal/gtk3",
         "paths": {
-            "astal": ["/usr/share/astal/gjs"],
-            "astal/*": ["/usr/share/astal/gjs/*"]
         }
-    },
-    "exclude": ["astal/gtk3/jsx-runtime.ts"]
+    }
 }

